docs/user/discover.asciidoc

```diff
@@ -110,6 +110,38 @@ image:images/document-table.png[Document table with fields for manufacturer, geo
 . To rearrange the table columns, hover the mouse over a
 column header, and then use the move and sort controls.
 
+[float]
+[[add-field-in-discover]]
+=== Add a field
+
+What happens if you forgot to define an important value as a separate field? Or, what if you
+want to combine two fields and treat them as one?
+You can add a field to your index pattern from inside of **Discover**,
+and then use that field for analysis and visualizations,
+the same way you do with other fields.
+
+. Click the ellipsis icon (...), and then click *Add field to index pattern*.
++
+[role="screenshot"]
+image:images/add-field-to-pattern.png[Dropdown menu located next to index pattern field with item for adding a field to an index pattern, width=50%]
+
+. In the *Create field* form, enter `hello` for the name.
+
+. Turn on *Set value*.
+
+. Use the Painless scripting language to define the field:
```

Comment:
Think it would be good to add a link to the Painless docs here: 
https://www.elastic.co/guide/en/elasticsearch/reference/master/modules-scripting-painless.html